partner_relations/models/res_partner_relation_type_selection.py

```diff
@@ -1,149 +1,128 @@
 # -*- coding: utf-8 -*-
 # © 2014-2016 Therp BV <http://therp.nl>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
-'''
-Created on 23 may 2014
-
-@author: Ronald Portier, Therp
-
-rportier@therp.nl
-http://www.therp.nl
-
+"""
 For the model defined here _auto is set to False to prevent creating a
-database file. All i/o operations are overridden to use a sql SELECT that
-takes data from res_partner_connection_type where each type is included in the
+database file. The model is based on a SQL view based on
+res_partner_relation_type where each type is included in the
 result set twice, so it appears that the connection type and the inverse
 type are separate records..
 
 The original function _auto_init is still called because this function
 normally (if _auto == True) not only creates the db tables, but it also takes
 care of registering all fields in ir_model_fields. This is needed to make
 the field labels translatable.
-
-example content for last lines of _statement:
-select id, record_type,
-  customer_id, customer_name, customer_city, customer_zip, customer_street,
-  caller_id, caller_name, caller_phone, caller_fax, caller_email
-from FULL_LIST as ResPartnerRelationTypeSelection where record_type = 'c'
-ORDER BY ResPartnerRelationTypeSelection.customer_name asc,
-ResPartnerRelationTypeSelection.caller_name asc;
-
-'''
+"""
 from psycopg2.extensions import AsIs
+
 from openerp import api, fields, models
 from openerp.tools import drop_view_if_exists
+
 from .res_partner_relation_type import ResPartnerRelationType
-from .res_partner import PADDING
+
+
+PADDING = 10
```

Comment:
The padding (or multiplication) of res_partner_relation_type_selection and res_partner_relation_all are not completely dependent of each other, and there is no intrinsic reason they should be the same. For instance I could have excluded the symmetric types from both the record type 'a' and 'b' records and have a separate UNION to include those.

Changing the value to 2 will work because now we have two files (actually two times the same file), so for instance the normal types will then always be even and the inverse relations always odd. I adapted this technique from a case where there was a union of 3 files. I called it PADDING because basically all normal types get a '0' stuffed on the right side, and all inverse types a '1'. Using multiplication just was the easiest way to do this.

I think we will be hugely succesfull with our system if we hit the MAXINT/10 limit. Although I will think Odoo performance problems might actually somewhat hinder reaching that target.

I have no problem with changing the names of the variables, of their values. Bit I think it would be logically wrong to include the one for the other.